File: Army(c++)/Warriors/Vampire.cpp
#include "Vampire.hpp"

Vampire::Vampire(const std::string& name, int hp, int dmg) 
    : Unit(name, hp, dmg, new VampireState(this))
    , warrior(WarriorID::VAMPIRE) {}

Vampire::~Vampire() {}

WarriorID Vampire::getWarriorID() const {
    return this->warrior;
}

void Vampire::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

File: Army(java)/Warriors/Vampire.java
package Warriors;

import EnumID.*;
import Basic.*;
import State.*;

public class Vampire extends Unit {
    private WarriorID warrior;
    
    public Vampire(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.VAMPIRE;
        this.setState(new VampireState(this));
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }
}
